### PR TITLE
[4.3] Convert content contact plugin to services

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts the content contact plugin to services.

### Testing Instructions
- Enable the plugin
- Set link to contact in the article options to yes in the article tab
- Create a contact with assigned to the current logged in user
- Create an article and set featured to yes
- Open the article on the front

### Actual result BEFORE applying this Pull Request
The "Written by" information links to the contact.

### Expected result AFTER applying this Pull Request
The "Written by" information links to the contact.

### CODE REVIEW

File: plugins/content/contact/services/provider.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+return new class implements ServiceProviderInterface
+{
+    /**
+     * Registers the service provider with a DI container.
+     *
+     * @param   Container  $container  The DI container.
+     *
+     * @return  void
+     *
+     * @since   4.2.0
```
Comment (brianteeman) [inline plugins/content/contact/services/provider.php]: deploy?

Comment (laoneo) [reply plugins/content/contact/services/provider.php]: thanks

File: plugins/content/contact/src/Extension/Contact.php [hunk 2]
```diff
@@ -23,14 +23,9 @@
  *
  * @since  3.2
  */
-class PlgContentContact extends CMSPlugin
+class Contact extends CMSPlugin
```
Comment (brianteeman) [inline plugins/content/contact/src/Extension/Contact.php]: does this need to be declared final now?

Comment (laoneo) [reply plugins/content/contact/src/Extension/Contact.php]: Yes

**Comment by MacJoom** [comment]
I have tested this item :white_check_mark: successfully on 356b6fb304c94aa0365d7701b611625f66de5e9f<br /><br />Tested successfully on 4.3-dev - no difference<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38446">issues.joomla.org/tracker/joomla-cms/38446</a>.</sub>

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on 356b6fb304c94aa0365d7701b611625f66de5e9f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38446">issues.joomla.org/tracker/joomla-cms/38446</a>.</sub>

**Comment by obuisard** [comment]
Thank you again for the PR Allon @laoneo

**Pull request merged**